tests/Functional/ConfigurationTest.php

```diff
@@ -138,6 +138,55 @@ public function testChangeComponentLibrary(): void {
     }
   }
 
+  /**
+   * Test that the correct layout is used after changing theme template setting.
+   */
+  public function testChangeEclTemplate(): void {
+    foreach (['oe_theme', 'oe_theme_subtheme_test'] as $active_theme) {
+      $this->container->get('theme_handler')->setDefault($active_theme);
+      $this->container->set('theme.registry', NULL);
+
+      $page = $this->getSession()->getPage();
```

Comment:
I don't think this variable is relevant. You do 4 times $this->drupalGet() after you save the current page in a variable.